purview/utils/general.py

```diff
@@ -191,6 +200,58 @@ def query_neo4j(query):
                 if node_label not in results_dict:
                     results_dict[node_label] = []
                 serialized_node = inflated_node.serialized
+                serialized_node['resource_type'] = node_label
                 if serialized_node not in results_dict[node_label]:
-                    results_dict[node_label].append(inflated_node.serialized)
+                    results_dict[node_label].append(serialized_node)
     return results_dict
+
+
+def get_corelated_nodes(results):
+    """
+    Create a raw cypher query for story nodes and get a count of the nodes co-related to them.
+
+    :param dict results: a dictionary containing story nodes
+    :return nodes_count_dict: a dictionary containing counts of all co-related nodes from Neo4j
+    :rtype: dict
+    """
+    # To avoid circular imports
+    from purview.models import story_flow
+
+    queries_dict = {}
+    for label, artifact in results.items():
+        query = 'MATCH '
+        backward_label = story_flow[label]['backward_label']
+        if backward_label:
+            backward_rel = story_flow[label]['backward_relationship'][:-1]
+            uid_name = story_flow[label]['uid_name']
+            node = create_node_subquery(label, uid_name,
+                                        artifact[0][uid_name])
+            next_node = create_node_subquery(backward_label)
+            query += '{0}-[:{1}]-{2}\n'.format(node, backward_rel, next_node)
+            query += 'RETURN COUNT({0}) AS count'.format(
+                backward_label.lower())
+
+            queries_dict[backward_label] = query
+
+    nodes_count_dict = {}
+    for label, query in queries_dict.items():
```

Comment:
Rather than even having the `queries_dict`, I think you could just directly execute the Cypher in the previous loop (line 234 right now) and create `nodes_count_dict` as you go.

Reply:
Nice idea, I'll make the change :)